From c4753f76a30a3e96c573d8e863587ba3e686348e Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Wed, 29 Mar 2017 10:57:23 +0200
Subject: migcompat/rtl8139: Work around version bump

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20170329105723.7789-2-dgilbert@redhat.com>
Patchwork-id: 74581
O-Subject: [RHEL-7.4 qemu-kvm-rhev PATCH v2 1/1] migcompat/rtl8139: Work around version bump
Bugzilla: 1420195
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

commit 46fe8bef in 2.7 bumped the version number of the rtl8139
vmstate, and added back a field that had been lost ~7 years ago
by 9d29cde in v0.11.

To keep backwards compatibility we can't bump the version, so push
the version number back down and remove the field that was added.
The field doesn't seem to be that significant, especially since we've
survived for 7 years with out it.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

Rebase notes (2.9.0):
- Patch rewritten

(cherry picked from commit 45ca263288bcbc98f36fae68c55e1a1b55e09221)
---
 hw/net/rtl8139.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hw/net/rtl8139.c b/hw/net/rtl8139.c
index 450658c..80c62dc 100644
--- a/hw/net/rtl8139.c
+++ b/hw/net/rtl8139.c
@@ -3206,7 +3206,7 @@ static void rtl8139_pre_save(void *opaque)
 
 static const VMStateDescription vmstate_rtl8139 = {
     .name = "rtl8139",
-    .version_id = 5,
+    .version_id = 4,
     .minimum_version_id = 3,
     .post_load = rtl8139_post_load,
     .pre_save  = rtl8139_pre_save,
@@ -3287,7 +3287,9 @@ static const VMStateDescription vmstate_rtl8139 = {
         VMSTATE_UINT32(tally_counters.TxMCol, RTL8139State),
         VMSTATE_UINT64(tally_counters.RxOkPhy, RTL8139State),
         VMSTATE_UINT64(tally_counters.RxOkBrd, RTL8139State),
+#if 0 /* Disabled for Red Hat Enterprise Linux bz 1420195 */
         VMSTATE_UINT32_V(tally_counters.RxOkMul, RTL8139State, 5),
+#endif
         VMSTATE_UINT16(tally_counters.TxAbt, RTL8139State),
         VMSTATE_UINT16(tally_counters.TxUndrn, RTL8139State),
 
-- 
1.8.3.1

